R/shinyapp.R

```diff
@@ -296,30 +296,33 @@ initAutoReloadMonitor <- function(dir) {
   )
 
   lastValue <- NULL
-  observeLabel <- paste0("File Auto-Reload - '", basename(dir), "'")
-  obs <- observe(label = observeLabel, {
-    files <- sort_c(
-      list.files(dir, pattern = filePattern, recursive = TRUE, ignore.case = TRUE)
+  check_for_update <- function(paths) {
+    paths <- grep(
+      filePattern,
+      paths,
+      ignore.case = TRUE,
+      value = TRUE
     )
-    times <- file.info(files)$mtime
-    names(times) <- files
-
-    if (is.null(lastValue)) {
-      # First run
-      lastValue <<- times
-    } else if (!identical(lastValue, times)) {
-      # We've changed!
-      lastValue <<- times
-      cachedAutoReloadMostRecentChange(times)
-      autoReloadCallbacks$invoke()
+
+    if (length(paths) == 0) {
+      return()
     }
 
-    invalidateLater(getOption("shiny.autoreload.interval", 500))
-  })
+    max_mtime <- max(file.info(paths)$mtime, na.rm = TRUE)
```

Comment:
I'd love to not have to do this, is there a reason we need to retrieve the times? Could `cachedAutoReloadMostRecentChange()` be based not on greatest mtime, but instead, on the last time a change was noticed?

Reply:
I think this is even a correctness issue. I think the basic idea here is, anytime any watched file is changed/added/removed, all cachedFuncWithFile instances are unconditionally invalidated. Otherwise, restoring to an older version of a file would trigger autoreload but not invalidate cachedFuncWithFile instances.

Reply:
Great point, I had thought of using the current time but thought it was too simple to work. Turns out it's the right kind of simple 😊

Reply:
Thanks for the suggestion, this ends up simplifying things nicely. Change made in 802dfd1119a2e7eb8bdac840098d7ff7db8d675d